py/desitarget/mock/build.py

```diff
@@ -1042,3 +1060,509 @@ def join_targets_truth(mockdir, outdir=None, force=False, comm=None):
         mtl.meta['EXTNAME'] = 'MTL'
         mtl.write(tmpout, overwrite=True, format='fits')
         os.rename(tmpout, out_mtl)
+
+def _get_magnitudes_onebrick(specargs):
+    """Filler function for the multiprocessing."""
+    return get_magnitudes_onebrick(*specargs)
+
+def get_magnitudes_onebrick(target_name, mockformat, thisbrick, brick_info, Magnitudes, dust_dir,
+                         select_targets_function, source_data, rand, log):
+    """Wrapper function to generate magnitudes for all the objects on a single brick."""
+
```

Comment:
I would prefer to keep the `magnitudes` vs. `spectra` functions completely separate. The `magnitudes` function doesn't actually need any speed-up.